components/bitcoin-topics/Navigation.tsx

```diff
@@ -40,7 +48,55 @@ export function Navigation({
     onLinkClick?: React.MouseEventHandler<HTMLAnchorElement>
 }) {
     const pathname = usePathname()
-    const posts = allCoreContent(sortPosts(allTopics))
+    const posts = useMemo(() => allCoreContent(sortPosts(allTopics)), [])
+    const [expandedTopics, setExpandedTopics] = useState<
+        Record<string, boolean>
+    >({})
+    const [lastVisitedTopic, setLastVisitedTopic] = useState<string | null>(
+        null
+    )
+
+    // Load expanded state from localStorage on component mount
+    useEffect(() => {
+        const savedState = localStorage.getItem("expandedTopics")
+        if (savedState) {
+            try {
+                const parsedState = JSON.parse(savedState)
+                setExpandedTopics(parsedState)
+            } catch (error) {
+                console.error(
+                    "Error parsing expandedTopics from localStorage:",
+                    error
+                )
+            }
+        }
+    }, [])
+
+    // Save expanded state to localStorage whenever it changes
+    useEffect(() => {
+        localStorage.setItem("expandedTopics", JSON.stringify(expandedTopics))
+        // Store expanded state for children as well
+        const expandedChildren = Object.keys(expandedTopics).reduce(
+            (acc, key) => {
+                const children = expandedTopics[key]
+                    ? navigation
+                          .find((section) =>
+                              section.links.some((link) => link.href === key)
+                          )
+                          ?.links.find((link) => link.href === key)?.children
+                    : []
+                if (children) {
+                    acc[key] = children.map((child) => child.href)
+                }
+                return acc
+            },
+            {} as Record<string, string[]>
+        )
+        localStorage.setItem(
+            "expandedChildren",
+            JSON.stringify(expandedChildren)
+        )
+    }, [expandedTopics])
```

Comment:
Github actions is suggesting adding 'navigation' as a dependency in the dependency array, you might want to check it out

Reply:
can you point me to the github action log